### PR TITLE
NOT SO NAÏVE

### PR DESCRIPTION
**Pull request by DavideB45**
naive bayes:
- classic
- with negation
- with stop words
the classic is the best one at about 77% accuracy

### CODE REVIEW

File: count_schools.py
Comment (M4rinz) [inline count_schools.py:1]: Cos'è? Va incorporato nell'esplorazione dei dati?

File: NaiveBayes/naiveBaiesModel.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    #x_train, x_test, y_train, y_test = splitData(addNegationsToData(getData()))
+    x_train, x_test, y_train, y_test = splitData(getData())
+    vectorizer = CountVectorizer(analyzer='word', stop_words=getStopWords())
+    #vectorizer = CountVectorizer()
+    x_train = vectorizer.fit_transform(x_train)
+    x_test = vectorizer.transform(x_test)
+
+    # fitprior (to set whether to learn class prior probabilities or not)
+    # alpha (smoothing parameter)
+    model = MultinomialNB(alpha=0.05, fit_prior=True)
```
Comment (M4rinz) [inline NaiveBayes/naiveBaiesModel.py]: alpha=0.05 fa lo smoothing? Scelto con grid search/trial and error?

File: NaiveBayes/naiveBaiesModel.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ############################################
+    ########### End of the main code ###########
+    ############################################
+    
+    cm = confusion_matrix(y_test, pred, labels=SCHOOLS, normalize='true')
+    disp = ConfusionMatrixDisplay(confusion_matrix=cm, display_labels=SCHOOLS)
+    disp.plot()
+    plt.xticks(rotation=45)
+    plt.show()
+    print(model.score(x_train, y_train))
```
Comment (M4rinz) [inline NaiveBayes/naiveBaiesModel.py:70]: Per curiosità, come viene fatto lo scoring?

File: NaiveBayes/naiveBaiesModel.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    model = MultinomialNB(alpha=0.05, fit_prior=True)
+    model.fit(x_train, y_train)
+    pred = model.predict(x_test)
+
+    ############################################
+    ########### End of the main code ###########
+    ############################################
+    
+    cm = confusion_matrix(y_test, pred, labels=SCHOOLS, normalize='true')
+    disp = ConfusionMatrixDisplay(confusion_matrix=cm, display_labels=SCHOOLS)
```
Comment (M4rinz) [inline NaiveBayes/naiveBaiesModel.py:62]: Viene salvata un'immagine di questa bellissima confusion matrix?

**Review by M4rinz — COMMENTED** [review]
Quasi approvata per me :D (sono molto puntiglioso)

**Comment by giordanoscerra** [comment]
Ue uagliu ma quante pull request

File: NaiveBayes/naiveBaiesModel.py (hunk 1 above)
Comment (DavideB45) [reply NaiveBayes/naiveBaiesModel.py]: hand made grid search

File: NaiveBayes/naiveBaiesModel.py (hunk 3 above)
Comment (DavideB45) [reply NaiveBayes/naiveBaiesModel.py:62]: no, appena servono le salvo a mano, salvo solo quelle che decidiamo di mettere nella presentazione

File: NaiveBayes/naiveBaiesModel.py (hunk 2 above)
Comment (DavideB45) [reply NaiveBayes/naiveBaiesModel.py:70]: per ora solo accuracy, devo aggiungere f1 e tutte le altre metriche, direi di creare una funzione in un file utils in modo che ci fa tutti i plot e calcoli le stesse metriche in modo da utilizzare tutti la stessa cosa

File: count_schools.py
Comment (DavideB45) [reply count_schools.py:1]: se vuoi lo buttiamo

Comment (DavideB45) [reply count_schools.py:1]: no questo, iil file sopra, questo credo possa essere utile se qualcuno deve fare delle rapide misurazioni

Comment (M4rinz) [reply count_schools.py:1]: Ok

File: NaiveBayes/naiveBaiesModel.py (hunk 2 above)
Comment (M4rinz) [reply NaiveBayes/naiveBaiesModel.py:70]: Ho visto `scoring.py`. Se non sbaglio `sklearn.classification_report()` stampa tante belle cosine (ieri le abbiamo un po' viste), il problema è semmai identificare che ci serve... Giusto?
Va bene anche così per ora forse, semmai ci porremo di nuovo il problema della valutazione (sia quali "punteggi" valutare, sia come ottenerli...) quando avremo altri modelli

File: NaiveBayes/naiveBaiesModel.py (hunk 3 above)
Comment (M4rinz) [reply NaiveBayes/naiveBaiesModel.py:62]: La `scorePhilosophy()` salva anche giusto? Se così direi di chiudere questa conversazione e fare il merge (mi sa che al momento in cui scrivo ci sono dei conflitti...)

Comment (M4rinz) [reply NaiveBayes/naiveBaiesModel.py:62]: Ho visto che la `ScorePhilosophy()` fa un po' di cosine. Bella. Chiudo la conversazione. Ciao
